controller/server/main.go

```diff
@@ -52,12 +56,17 @@ func init() {
 type server struct {
 	cpb.UnimplementedTopologyManagerServer
 
-	mu          sync.Mutex
+	muDeploy    sync.Mutex // guards deployements map
 	deployments map[string]*deploy.Deployment
+	muTopo      sync.Mutex        // guards topos map
+	topos       map[string][]byte // stores the topology protobuf from the initial topology creation request
```

Comment:
Is there a reason to store the raw bytes instead of having the map be of type map[string]*tpb.Topology ?

Reply:
In current implementation of Create/Delete/Show topology APIs, the input *tpb.Topology will be modified after the call is done. i.e., this topology pb will be modified by the TopologyManager's Load() method. 

Several ways to fix this problem:
1) Change the current API implementation: instead of getting and loading a protobuf (topo.Load() and then calling manager.Load()), we can passed a "loaded" topology protobuf.
2) Without changing the current API implementation, store it as a *tpb.Topology and the callers need to have a deep copy of it when using it.

Storing it as raw bytes is the simplest way to work around this issue... Any thoughts on this?

Reply:
Lets discuss this at our meeting today